rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java

```diff
@@ -102,10 +104,30 @@ public Web3RskImpl(
         this.blockStore = blockStore;
     }
 
-    public void ext_dumpState() {
+    public void ext_dumpState(String account,boolean exportStorageKeys,boolean exportCode) {
         Block bestBlcock = blockStore.getBestBlock();
-        logger.info("Dumping state for block hash {}, block number {}", bestBlcock.getHash(), bestBlcock.getNumber());
-        networkStateExporter.exportStatus(System.getProperty("user.dir") + "/" + "rskdump.json");
+        logger.info("Dumping state for block hash {}, block number {}, account {}",
+                bestBlcock.getHash(), bestBlcock.getNumber(),account);
+        String name = "rskdump";
+        if (account.length()!=0) {
+            RskAddress addrToCheck;
+            try {
+                addrToCheck = new RskAddress(account);
+
+                // Now make sure we don't enable path-traversal vulnerabilities in case
+                // the address format is ever changed.
+                if (account.indexOf(File.separatorChar)>=0) {
```

Comment:
is this check needed since hex representation of address `addrToCheck.toHexString()` is used for filename?

Reply:
As the comment explain, It's a second layer of defense in case in the future the RSK address format allows addresses containing '/' or '\'.

Reply:
In case new format contains a separator char and `addrToCheck.toHexString()` is still used for filename it's ok not to check the separator (as it would be encoded as `2f` or `5c`). In case other encoding of address will be used for filename than yes it may require a check (but this also may become a problem as legitimate addresses containing a separator wouldn't pass the check).

Reply:
You may be totally right, but since the code is used only for testing, I think our time discussing this issue is more valuable than the cost of editing the line of code, and the benefit it provides.
So, I'm happy with either removing or leaving the check.